azure/scope/machinepool.go

```diff
@@ -19,8 +19,11 @@ package scope
 import (
 	"context"
 	"encoding/base64"
+	"fmt"
+	"strings"
 	"time"
 
+	azure2 "github.com/Azure/go-autorest/autorest/azure"
```

Comment:
do we have a better convention for naming this anywhere else in the repo already? azure2 isn't very explicit. If so this would be a good one to add to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.golangci.yml#L59

Reply:
See #2958